WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -96,6 +100,14 @@ def index
       @competitions = @competitions.select { |competition| competition.has_events_with_ids?(params[:event_ids]) }
     end
 
+    unless params[:status] == "all"
+      if params[:status] == "warning"
+        @competitions = @competitions.select { |competition| competition.pending_results_or_report(14) }
+      else
+        @competitions = @competitions.select { |competition| competition.pending_results_or_report(21) }
```

Comment:
How are we going to remember to keep this code in sync with [`report_class`](https://github.com/cubing/worldcubeassociation.org/blob/3231eeb3e68e47d15f8b5e57fcc46de359e7b3c0/WcaOnRails/app/helpers/competitions_helper.rb#L23) and [`results_class`](https://github.com/cubing/worldcubeassociation.org/blob/3231eeb3e68e47d15f8b5e57fcc46de359e7b3c0/WcaOnRails/app/helpers/competitions_helper.rb#L28)?

Also, what do you plan to do if someone changes `results_class` so it does something different than `report_class`?